docs/reference/namespaces.md

```diff
@@ -0,0 +1,174 @@
+    plugins: [database0, blockchain0, dataexchange0, sharedstorage0, erc20_erc721]
+    multiparty:
+      networkNamespace: alpha
+      enabled: true
+      org:
+        name: org0
+        description: org0
+        key: 0x123456
+      node:
+        name: node0
+        description: node0
+      contract:
+        - location:
+            address: 0x4ae50189462b0e5d52285f59929d037f790771a6
+          firstEvent: 0
+        - location:
+            address: 0x3c1bef20a7858f5c2f78bda60796758d7cafff27
+          firstEvent: 5000
+  - name: omega
+    defaultkey: 0x48a54f9964d7ceede2d6a8b451bf7ad300c7b09f
+    description: Gateway namespace
+    plugins: [database0, blockchain1, erc20_erc721]
+```
+
+The `namespaces.predefined` object contains the follow sub-keys:
+
+* `defaultKey` is a blockchain key used to sign transactions when none is specified (in multi-party mode,
+  defaults to the org key)
+* `plugins` is an array of plugin names to be activated for this namespace (defaults to
+  all available plugins if omitted)
+* `multiparty.networkNamespace` is the namespace name to be sent in plugin calls, if it differs from the
+  locally used name (useful for interacting with multiple shared namespaces of the same name -
+  defaults to the value of `name`)
+* `multiparty.enabled` controls if multi-party mode is enabled (defaults to true if an org key or
+  org name is defined on this namespace _or_ in the deprecated `org` section at the root)
+* `multiparty.org` is the root org identity for this multi-party namespace (containing `name`,
+  `description`, and `key`)
+* `multiparty.node` is the local node identity for this multi-party namespace (containing `name` and
+  `description`)
+* `multiparty.contract` is an array of objects describing the location(s) of a FireFly multi-party
+  smart contract. Its children are blockchain-agnostic `location` and `firstEvent` fields, with formats
+  identical to the same fields on custom contract interfaces and contract listeners. The blockchain plugin
+  will interact with the first contract in the list until instructions are received to terminate it and
+  migrate to the next.
+
+### Config Restrictions
+* `name` must be unique on this node
+* for historical reasons, "ff_system" is a reserved string and cannot be used as a `name` or `multiparty.networkNamespace`
+* a `database` plugin is required for every namespace
+* if `multiparty.enabled` is true, plugins _must_ include one each of `blockchain`, `dataexchange`, and
+  `sharedstorage`
+* if `multiparty.enabled` is false, plugins _must not_ include `dataexchange` or `sharedstorage`
+* at most one of each type of plugin is allowed per namespace, except for tokens (which
+  may have many per namespace)
+
+All namespaces must be called out in the FireFly config file in order to be valid. Namespaces found in
+the database but _not_ represented in the config file will be ignored.
+
+## Local Definitions
+
+The following are all "definition" types in FireFly:
```

Comment:
Think a `## Definitions` section is needed with a paragraph describing what a definition is, before the section where we specify which are local. Because it's not a core construct on the API or anything - it's only really at the moment exposed in the code architecture